src/linear_algebra/multilinear/basis.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2021 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import linear_algebra.basis
+import linear_algebra.multilinear.basic
+
+/-!
+# Multilinear maps in relation to bases.
+
+This file proves lemmas about the action of multilinear maps on basis vectors.
+
+-/
+
+open multilinear_map
+
+variables {R : Type*} {ι : Type*} {n : ℕ} {M : fin n.succ → Type*} {M₂ : Type*} {M₃ : Type*}
+variables [comm_semiring R] [add_comm_monoid M₂] [add_comm_monoid M₃] [∀i, add_comm_monoid (M i)]
+variables [∀i, module R (M i)] [module R M₂] [module R M₃]
+
+/-- Two multilinear maps indexed by `fin n.succ` are equal if they are equal when all arguments
+are basis vectors. -/
+lemma basis.ext_multilinear_fin {f g : multilinear_map R M M₂} {ι₁ : fin n.succ → Type*}
+  (e : Π i, basis (ι₁ i) R (M i)) (h : ∀ v : Π i, ι₁ i, f (λ i, e i (v i)) = g (λ i, e i (v i))) :
+  f = g :=
+begin
+  unfreezingI { induction n with m hm },
+  { rw [←f.uncurry_curry_left, ←g.uncurry_curry_left],
+    congr' 1,
+    refine basis.ext (e 0) _,
+    intro i,
+    ext v,
+    rw [curry_left_apply, curry_left_apply],
+    convert h (fin.cons i (λ x, fin_zero_elim x)),
+    { ext x,
+      have h0 : x = 0 := subsingleton.elim _ _,
+      rw h0,
+      simp },
+    { ext x,
+      have h0 : x = 0 := subsingleton.elim _ _,
+      rw h0,
+      simp } },
+  { rw [←f.uncurry_curry_left, ←g.uncurry_curry_left],
+    congr' 1,
```

Comment:
```suggestion
  { apply function.left_inverse.injective uncurry_curry_left,
```

Reply:
I've applied that change (I've not done anything about adding that separate lemma).